Vagrantfile

```diff
@@ -35,20 +39,53 @@ Vagrant.configure(2) do |config|
 
   config.vm.synced_folder ".", "/vagrant", type: "nfs"
 
+  domain_suffix = "demo"
   machines.each do |machine|
-    config.vm.define machine[:name] do |vmconfig|
-      vmconfig.vm.box = "bento/centos-6.7"
+    short_name = machine[:name]
+    domain_name = "#{short_name}.#{domain_suffix}"
+
+    config.vm.define short_name do |vmconfig|
+      vmconfig.vm.box = BOX_NAME
       vmconfig.vm.provider "virtualbox" do |v|
+        v.name = short_name
         v.memory = machine[:memory]
         v.cpus = `#{RbConfig::CONFIG['host_os'] =~ /darwin/ ? 'sysctl -n hw.ncpu' : 'nproc'}`.chomp
       end
 
-      vmconfig.vm.hostname = machine[:name]
+      vmconfig.vm.hostname = domain_name
       vmconfig.vm.network :private_network, ip: machine[:ip]
 
       vmconfig.vm.provision "ansible" do |ansible|
-        ansible.playbook = machine[:ansible]
+        ansible.playbook = ANSIBLE_PLAYBOOK
+        ansible.verbose = 'vvvv'
+
+        # Use helper method to define machine groups
+        ansible.groups = group_up(machines)
+      end
+    end
+  end
+end
+
+# Helper method to define machine groupings
+def group_up (machines)
```

Comment:
can you add a comment to describe what this method actually does? Why it is actually necessary?